client/src/helpers/fhir/fhir.js

```diff
@@ -523,6 +523,24 @@ export const createPatientSubmissionBundle = ({
   return bundle;
 };
 
+export const createServiceRequestResource = (
+  practitionerId, patient, serviceRequestCode, newStatus, serviceRequestId,
+) => {
+  const patientEntry = createEntry(patient);
+  const patientReference = { reference: patient.id ? `Patient/${patient.id}` : patientEntry.fullUrl };
+  const serviceRequestResource = FhirDataManager.createServiceRequest(
+    practitionerId,
+    patientEntry.fullUrl,
+    newStatus,
+    serviceRequestCode,
+  );
+
+  serviceRequestResource.id = serviceRequestId || undefined;
+  serviceRequestResource.subject = patientReference;
+
+  return serviceRequestResource;
+};
```

Comment:
How about doing something like: 
```
const serviceRequestResource = FhirDataManager.createServiceRequest(
    practitionerId,
    patient.id,
    newStatus,
    serviceRequestCode,
  );
  serviceRequestResource.id = serviceRequestId || undefined;
```
This way, you don't even need to have the function